tests/test_plan.py

```diff
@@ -977,23 +941,11 @@ def generate_mocked_context(prefix, root_dir, envs_dirs):
 
 class TestDetermineAllEnvs(unittest.TestCase):
     def setUp(self):
-        pkgs = {
-            "test-spec": [generate_mocked_package("test1", "test-spec", "defaults", "1"),
-                          generate_mocked_package(None, "test-spec", "defaults", "3"),
-                          generate_mocked_package(None, "test-spec", "defaults", "2"),
-                          generate_mocked_package("ranenv", "test-spec", "rando_chnl", "5")],
-            "test-spec2": [generate_mocked_package(None, "test-spec2", "defaults", "1")],
-            "no-exist": [generate_mocked_package(None, "no-exist", "nope", "1")]
-        }
-        index = {
-            Dist.from_string("test-spec", channel_override="defaults"):
-                generate_mocked_package(None, "test-spec", "default", "1"),
```

Comment:
Note that the channel here is "default", not "defaults". It's not clear that inconsistency was ever going to be a problem, but it's one reason why I think it's better to generate the mocked data all at once, not in two passes like this.